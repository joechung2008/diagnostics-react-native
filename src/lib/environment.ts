export type Environment = (typeof Environment)[keyof typeof Environment];

export const Environment = {
  Public: 'https://hosting.portal.azure.net/api/diagnostics',
  Fairfax: 'https://hosting.azureportal.usgovcloudapi.net/api/diagnostics',
  Mooncake: 'https://hosting.azureportal.chinacloudapi.cn/api/diagnostics',
} as const;

export function getEnvironmentName(
  environment: Environment | undefined,
): string {
  switch (environment) {
    case Environment.Public:
      return 'Public Cloud';
    case Environment.Fairfax:
      return 'Fairfax';
    case Environment.Mooncake:
      return 'Mooncake';
    default:
      return 'Select environment';
  }
}
