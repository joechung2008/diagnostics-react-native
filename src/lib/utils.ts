export function isExtensionInfo(
  value: Extension | undefined,
): value is ExtensionInfo {
  return value !== undefined && 'extensionName' in value;
}

export function byKey(a: KeyedNavLink, b: KeyedNavLink): number {
  return a.key < b.key ? -1 : a.key > b.key ? 1 : 0;
}

export function toNavLink({ extensionName }: ExtensionInfo): KeyedNavLink {
  return {
    key: extensionName,
    name: extensionName,
    url: '',
  };
}

export function when<T>(condition: boolean, ...args: T[]): T[] {
  return condition ? args : [];
}

export async function fetchDiagnostics(
  environment: string,
): Promise<Diagnostics> {
  const response = await fetch(environment);
  if (!response.ok) {
    throw new Error(`Failed to fetch diagnostics: ${response.statusText}`);
  }
  return response.json();
}
