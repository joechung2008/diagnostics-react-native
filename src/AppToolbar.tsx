import { useAppToolbarStyles } from '@/hooks/useAppToolbarStyles';
import { Environment, getEnvironmentName } from '@/lib/environment';
import { useTheme } from '@/ThemeContext';
import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';

interface AppToolbarProps {
  environment: Environment;
  setShowEnvironmentPicker: (show: boolean) => void;
  showEnvironmentPicker: boolean;
  showPaasServerless: boolean;
  handleEnvironmentChange: (value: Environment) => void;
  handlePaasServerlessPress: () => void;
  handleWebsitesPress: () => void;
}

const AppToolbar: React.FC<AppToolbarProps> = ({
  environment,
  setShowEnvironmentPicker,
  showEnvironmentPicker,
  showPaasServerless,
  handleEnvironmentChange,
  handlePaasServerlessPress,
  handleWebsitesPress,
}) => {
  const { colors } = useTheme();
  const dynamicStyles = useAppToolbarStyles(colors);

  return (
    <>
      <View style={dynamicStyles.toolbar}>
        <TouchableOpacity
          style={dynamicStyles.environmentButton}
          onPress={() => setShowEnvironmentPicker(!showEnvironmentPicker)}
        >
          <Text style={dynamicStyles.environmentButtonText}>
            {getEnvironmentName(environment)} ðŸ”½
          </Text>
        </TouchableOpacity>
        <View style={dynamicStyles.toolbarRight}>
          {showPaasServerless && (
            <TouchableOpacity
              style={dynamicStyles.toolbarButton}
              onPress={handlePaasServerlessPress}
            >
              <Text style={dynamicStyles.toolbarButtonText}>
                paasserverless
              </Text>
            </TouchableOpacity>
          )}
          <TouchableOpacity
            style={dynamicStyles.toolbarButton}
            onPress={handleWebsitesPress}
          >
            <Text style={dynamicStyles.toolbarButtonText}>websites</Text>
          </TouchableOpacity>
        </View>
      </View>

      {showEnvironmentPicker && (
        <>
          <TouchableOpacity
            style={dynamicStyles.dropdownOverlay}
            onPress={() => setShowEnvironmentPicker(false)}
            activeOpacity={1}
          />
          <View style={dynamicStyles.environmentDropdown}>
            {Object.entries(Environment).map(([key, value], index, array) => (
              <TouchableOpacity
                key={key}
                style={[
                  dynamicStyles.environmentOption,
                  index === array.length - 1 &&
                    dynamicStyles.environmentOptionLast,
                ]}
                onPress={() => handleEnvironmentChange(value)}
              >
                <Text style={dynamicStyles.environmentOptionText}>
                  {getEnvironmentName(value)}
                </Text>
              </TouchableOpacity>
            ))}
          </View>
        </>
      )}
    </>
  );
};

export default AppToolbar;
