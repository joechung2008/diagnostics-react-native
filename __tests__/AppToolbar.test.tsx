import AppToolbar from '@/AppToolbar';
import { Environment } from '@/lib/environment';
import { render } from '@testing-library/react-native';
import React from 'react';

// Mock the ThemeContext
jest.mock('@/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      text: '#212529',
      primary: '#007bff',
      surface: '#f8f9fa',
    },
  }),
}));

// Mock styles hook
jest.mock('@/hooks/useAppToolbarStyles', () => ({
  useAppToolbarStyles: () => ({
    toolbar: {},
    toolbarButton: {},
    environmentButton: {},
    environmentText: {},
    pickerContainer: {},
    pickerButton: {},
  }),
}));

describe('AppToolbar', () => {
  const defaultProps = {
    environment: Environment.Public,
    setShowEnvironmentPicker: jest.fn(),
    showEnvironmentPicker: false,
    showPaasServerless: false,
    handleEnvironmentChange: jest.fn(),
    handlePaasServerlessPress: jest.fn(),
    handleWebsitesPress: jest.fn(),
  };

  it('renders correctly with default props', () => {
    const { toJSON } = render(<AppToolbar {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with environment picker shown', () => {
    const { toJSON } = render(
      <AppToolbar {...defaultProps} showEnvironmentPicker={true} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with PaaS Serverless shown', () => {
    const { toJSON } = render(
      <AppToolbar {...defaultProps} showPaasServerless={true} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with different environment', () => {
    const { toJSON } = render(
      <AppToolbar {...defaultProps} environment={Environment.Fairfax} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
