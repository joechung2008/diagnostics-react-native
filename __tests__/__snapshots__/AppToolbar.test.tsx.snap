// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AppToolbar renders correctly with PaaS Serverless shown 1`] = `
<View
  style={{}}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text>
      Public Cloud
       ðŸ”½
    </Text>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        paasserverless
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        websites
      </Text>
    </View>
  </View>
</View>
`;

exports[`AppToolbar renders correctly with default props 1`] = `
<View
  style={{}}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text>
      Public Cloud
       ðŸ”½
    </Text>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        websites
      </Text>
    </View>
  </View>
</View>
`;

exports[`AppToolbar renders correctly with different environment 1`] = `
<View
  style={{}}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text>
      Fairfax
       ðŸ”½
    </Text>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        websites
      </Text>
    </View>
  </View>
</View>
`;

exports[`AppToolbar renders correctly with environment picker shown 1`] = `
[
  <View
    style={{}}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        Public Cloud
         ðŸ”½
      </Text>
    </View>
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text>
          websites
        </Text>
      </View>
    </View>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  />,
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        Public Cloud
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        Fairfax
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        Mooncake
      </Text>
    </View>
  </View>,
]
`;
